packages/app-staking/src/Account/index.tsx

```diff
@@ -58,18 +59,20 @@ class Account extends React.PureComponent<Props, State> {
 
   render () {
     const { accountId, controllerId, filter, name, stashId } = this.props;
+    const { isBondExtraOpen, isBondOpen, isNominateOpen, isSessionKeyOpen, isUnbondOpen, isValidatingOpen } = this.state;
 
     if ((filter === 'controller' && !stashId) || (filter === 'stash' && !controllerId) || (filter === 'unbonded' && (controllerId || stashId))) {
       return null;
     }
 
     return (
       <article className='staking--Account'>
-        {this.renderBond()}
-        {this.renderBondExtra()}
-        {this.renderNominating()}
-        {this.renderSessionKey()}
-        {this.renderValidating()}
+        {isBondOpen && this.renderBond()}
```

Comment:
Re-copying the comment here.

There are actually very good reasons why some of these are done this way with the passing through of `isOpen`.

In the cases where we do queries and display the info to the user, we want to fire them in the background, so there is no actual delay for the display of the info. It has the additional benefit that the subscriptions are not re-created each time as well.

(The latter probably only until we get multi support inside the API)

... and all obviously not obvious as to why

Reply:
Damn I answered on the original comment but it didn't get through.

Thanks for the info, very interesting. Before I revert it all you say "why **some** of these are done this way". Should I keep this "new" behaviour (without passing the isOpen) for some components?

Reply:
Hmm, but none of them are actually doing any queries (as in API queries) in the background, or am I missing something?
```
{this.renderBond()} -> src/Account/Bond.tsx
{this.renderBondExtra()} -> src/Account/BondExtra.tsx
{this.renderNominating()} -> src/Account/Nominating.tsx
{this.renderSessionKey()} -> src/Account/SessionKey.tsx
{this.renderValidating()} -> src/Account/Validating.tsx
```
The only component doing an API call in ./Accounts is the `ValidateController.tsx` which is used in `Account/Bond.tsx` and called from within `renderBond()` .

The way it is implemented we're doing an an api all in `ValidateController` for every account listed under the  'Account Actions' tab, even if the user never opens the Bond overlay. That would make one unneeded call for every account as opposed to maybe re-creating a subsription for the same account in case the user is opening the same overlay twice (or more)...